### DISCUSSION TITLE
What parameters could affect the Fe Berry curvature calculations

### DISCUSSION THREAD

**Discussion by Dongsheng-Wen** [Chengcheng-Xiao/VASP2WAN90_v2_fix#15] [Q&A]
Hi Chengcheng, 

Thank you for sharing the demo example. I was trying to reproduce the results for Fe Berry curvature. 
Instead of completely using the parameters you given in the vasp inputs, I changed the inputs and would like to see what might affect the results. 
For vasp: 
- I used a denser kmesh 20x20x20
- The cell was relaxed to a slightly different volume with EDIFFG=-1e-3.
- Number of bands are changed from 40 to 48. 
For wannier90, basically they are copied from your file except num_bands = 48 and the fermi energy was set based on my static run. 
```
num_bands = 48
num_wann = 18
spinors = .true.

dis_win_min       = -20.0d0
dis_win_max       = 70.0d0
dis_froz_min      = -20.0d0
dis_froz_max      =  15.0d0
dis_num_iter      =  500

num_iter          = 200
dis_mix_ratio = 1.0

begin projections
Fe: sp3d2;dxy;dxz;dyz
end projections
fermi_energy = 5.82347
kpath = true
kpath_task = bands+curv
kpath_bands_colour = none
kpath_num_points = 1000
begin kpoint_path
G 0.0000 0.0000 0.0000   H 0.500 -0.5000 -0.5000
H 0.500 -0.5000 -0.5000  P 0.7500 0.2500 -0.2500
P 0.7500 0.2500 -0.2500  N 0.5000 0.0000 -0.5000
N 0.5000 0.0000 -0.5000  G 0.0000 0.0000 0.000
G 0.0000 0.0000 0.000    H 0.5 0.5 0.5
H 0.5 0.5 0.5            N 0.5 0.0 0.0
N 0.5 0.0 0.0            G 0.0 0.0 0.0
G 0.0 0.0 0.0            P 0.75 0.25 -0.25
P 0.75 0.25 -0.25        N 0.5 0.0 0.0
end kpoint_path
```
Here is what I've got for the Berry curvature along the high symmetry paths. 
<img width="1113" alt="image" src="https://user-images.githubusercontent.com/91816327/201385684-5caf91e3-7f05-428c-b1e8-ef4c38be19bd.png">

The band structure seems to be reproduced but the Berry curvature is not the same. I wonder what could be the reason? 
If the discrepancies comes from VASP calculations, I think it might be the ```num_bands```, because the denser kmesh is more likely to be accurate. 

Best,
Dongsheng

**Comment by Chengcheng-Xiao** [comment]
Hum, I'm not entirely sure that result is wrong (as I'm not an expert on the subject). Have you tried running the same thing with QE?

**Comment by Dongsheng-Wen** [comment]
Okay. 
Then I carried out the kslice and kubo tasks and here are the results. 

For Fig.3: 
<img width="371" alt="image" src="https://user-images.githubusercontent.com/91816327/203077072-f465570d-5034-4266-aead-76155b5720f0.png">

For Fig.5 lower and upper subplots: 
![image](https://user-images.githubusercontent.com/91816327/203072496-328429ff-d815-4260-8a06-621c1320b475.png)
![image](https://user-images.githubusercontent.com/91816327/203072836-c7dbb276-8394-4539-b50d-e9c2bb7e9c95.png)

It looks like the general features (the peak positions and trends) were reproduced while the magnitudes are different. I double checked the reference which used wien2k with LAPW in the calculation. With no experiences in wien2k, I can only guess that the discrepancies came from the VASP (PAW) and wien2k(LAPW). 

Best,
Dongsheng

**Reply by Chengcheng-Xiao** [reply]
Hum, maybe give example 18 in Wannier90's tutorial a try?
